gcivil-nyu-org/fall2019-cs-gy-6063-team-debug_entities
# PR for #111: Restrict Visibility of "Edit Profile" Button

### Description

* I restricted visibility of the "Edit Profile" button so a user will only see that button on their own profile.
* The `/u/<id>` URLs were showing the current user's info, regardless of the number given in the URL i.e. you could never see anyone's info except your own. So I fixed that by adding the requested user to the context of the user() function in views.py. Then in user.html, we use the data from the context, rather than the current user's data.
* I removed the homemade authentication from user.html and the user() function in views.py because the decorator takes care of it for us
* I added a check to the edit_profile() function in views.py so that a user can't edit another user's profile
* I added an error message that's shown if you try to view the profile of an id that doesn't exist. You also can't view the profile of a user who hasn't verified their email address.

### Testing Directions

```
python manage.py makemigrations
python manage.py migrate
python manage.py pull_seatgeek_data
```

1. Get the code from this PR
1. Run the Django server. Make sure you have at least two users, I'll call them A and B.
1. Sign in as A and go to A's profile at `/u/<A's id>`. See that the "Edit Your Profile" button is visible. Click it and see that A can edit their own profile, as desired.
> ![image](https://user-images.githubusercontent.com/13837978/67536657-41c5d900-f6a6-11e9-914a-0b6a587f21a9.png)
1. While still signed in as A, go to B's profile at `/u/<B's id>`. See that the "Edit Your Profile" button is not shown, as desired. Also see that B's info is shown instead of A's, as desired.
> ![image](https://user-images.githubusercontent.com/13837978/67536675-4d190480-f6a6-11e9-822c-195a4c3b932e.png)
1. Try to edit B's profile anyway by going to `/u/<B's id>/edit`. See that you get a 403 error message, as desired, instead of being able to edit B's profile.
1. Go to a non-existent user profile at `/u/99999`. See that there's a 403 error message, as desired.
1. Make a new user and don't verify their email address. Sign in as a verified user and try to view the unverified user's profile. Get an error, as desired.

### Miscellaneous Comments

* The error messages that I've added when you go to a non-existent user profile or try to edit someone else's profile are super barebones 403 errors. If we want them to be pretty, let's do that in a separate issue.
* To make several users, it's useful to put a + in your email address. Anything after the +, and the + itself, will be ignored. For example, Vedanth's email address is `vd908@nyu.edu` but `vd908+1@nyu.edu` will still go to him.

### Checklist
